google/cloud/storage/internal/object_requests.h

```diff
@@ -125,6 +125,43 @@ class InsertObjectStreamingRequest
 std::ostream& operator<<(std::ostream& os,
                          InsertObjectStreamingRequest const& r);
 
+/**
+ * Represents a request to the `Objects: copy` API.
+ */
+class CopyObjectRequest
+    : public GenericRequest<
+          CopyObjectRequest, DestinationPredefinedAcl, IfGenerationMatch,
+          IfGenerationNotMatch, IfMetagenerationMatch, IfMetagenerationNotMatch,
+          IfSourceGenerationMatch, IfSourceGenerationNotMatch,
+          IfSourceMetagenerationMatch, IfSourceMetagenerationNotMatch,
+          Projection, SourceGeneration, UserProject> {
+ public:
+  CopyObjectRequest() = default;
+  CopyObjectRequest(std::string destination_bucket,
```

Comment:
Any reason the destination args are placed before the source args here?  Semantically, I'd expect source args to come before dest args.

Reply:
No particular reason, I have seen both: `std::memcpy()` using destination first, `std::copy()` using destination second. Happy to change it if you think it is more natural in the reverse order.

Reply:
Hmmm, I just realized that @frankyn change for compose  (#1090) uses destination first too. Let's keep both the same?

Reply:
No particular reason either. I don't mind reordering. I understand @houglum's point.

Reply:
I have reversed the order of the arguments